nmt.go

```diff
@@ -491,6 +501,27 @@ func (n *NamespacedMerkleTree) MaxNamespace() (namespace.ID, error) {
 	return MaxNamespace(r, n.NamespaceSize()), nil
 }
 
+// ForceAddLeaf adds a namespaced data to the tree without validating its
+// namespace ID. This method should only be used by tests that are attempting to
+// create out of order trees. The default hasher will fail for trees that are
+// out of order.
+func (n *NamespacedMerkleTree) ForceAddLeaf(leaf namespace.PrefixedData) error {
```

Comment:
> This method should only be used by tests

[question] is it possible to un-export this function and retain the tests that use it? It seems possible b/c `nmt_test.go` is in the same package



```suggestion
func (n *NamespacedMerkleTree) forceAddLeaf(leaf namespace.PrefixedData) error {
```

Asking so that we can enforce that users don't use this function

Reply:
^ disregard. I assume this PR is meant to be used in conjunction with subsequent PRs that do use this exported function

Reply:
yeah we have to use this in app to create malicious commitments by hand